momento-sdk/src/main/java/momento/sdk/CacheClient.java

```diff
@@ -677,6 +681,219 @@ public CompletableFuture<CacheSetFetchResponse> setFetch(String cacheName, Strin
+   * Creates the sorted set if it does not exist.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to add to.
+   * @param elements - The value to score map to add to the sorted set.
+   * @param ttl TTL for the set in cache. This TTL takes precedence over the TTL used when
+   *     initializing a cache client. Defaults to client TTL.
+   * @return Future containing the result of the put elements operation.
+   */
+  public CompletableFuture<CacheSortedSetPutElementsResponse> sortedSetPutElementsByteArray(
+      String cacheName,
+      String sortedSetName,
+      Map<byte[], Double> elements,
+      @Nullable CollectionTtl ttl) {
+    return scsDataClient.sortedSetPutElementsByteArray(cacheName, sortedSetName, elements, ttl);
+  }
+
+  /**
+   * Adds elements to the given sorted set. If the elements already exist, their scores are updated.
+   * * Creates the sorted set if it does not exist.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to add to.
+   * @param elements - The value to score map to add to the sorted set.
+   * @return Future containing the result of the put elements operation.
+   */
+  public CompletableFuture<CacheSortedSetPutElementsResponse> sortedSetPutElementsByteArray(
+      String cacheName, String sortedSetName, Map<byte[], Double> elements) {
+    return scsDataClient.sortedSetPutElementsByteArray(cacheName, sortedSetName, elements, null);
+  }
+
+  /**
+   * Fetch the elements in the given sorted set by index (rank).
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to fetch from.
+   * @param startRank - The rank of the first element to fetch. Defaults to 0. This rank is
+   *     inclusive, i.e. the element at this rank will be fetched.
+   * @param endRank - The rank of the last element to fetch. This rank is exclusive, ie the element
+   *     at this rank will not be fetched. Defaults to null, which fetches up until and including
+   *     the last element.
+   * @param order - The order to fetch the elements in. Defaults to ascending.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByRank(
+      String cacheName,
+      String sortedSetName,
+      @Nullable Integer startRank,
+      @Nullable Integer endRank,
+      @Nullable SortOrder order) {
+    return scsDataClient.sortedSetFetchByRank(cacheName, sortedSetName, startRank, endRank, order);
+  }
+
+  /**
+   * Fetch all elements in the given sorted set by index (rank) in ascending order.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to fetch from.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByRank(
```

Comment:
I added this and the equivalent sortedSetFetchByScore because I assume people will want to fetch the entire set regularly.

Reply:
👍